### DISCUSSION TITLE
Why is funder equal to msg.sender ?

### DISCUSSION THREAD

**Discussion by LavleshJ** [Cyfrin/foundry-full-course-cu#426] [Q&A] [answered]
After refactoring the code and using the script contract to return the FundMe object. Why is FundMe.i_owner() is equal to msg.sender ?
<img width="958" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/44592946/bddf1538-4b8e-446a-bd7b-5886d2d9de24">

<img width="475" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/44592946/fe9d685b-c082-4812-ba58-86825f2543a2">

It is showing in console log that both are equal, but logically how ?

<img width="653" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/44592946/c7b02b06-e093-4899-94de-726fb99477fc">

**Comment by shikhar229169** [comment, marked as answer]
When we write vm.startBroadcast(), we say that the next txn should be performed by that account for which we passed the --private-key argument in command line, but during test, we don't pass it, so what foundry does is, it automatically takes a account with fake eth and passes the private key internally, so the contract was deployed by that particular account. And coming to the msg.sender stuff over here, as the caller of the test function is again the account for which the --private-key arg is passed, (but for test foundry passes it automatically, we don't define that). Thus they both are same.

**Reply by LavleshJ** [reply]
Ah ! That makes sense. So, whenever there is ` vm.startBroadcast() `, the txn within its scope will be performed by the account passed while calling the forge ?

**Reply by LavleshJ** [reply]
Perfect ! The Mock contract also had `vm.startBroadcast()`, so caller is same.
<img width="832" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/44592946/a169ea71-74c4-4aab-bc9c-e159015651a0">
